chapters/packages.tex

```diff
@@ -321,3 +321,183 @@ \section{External Resources}\label{external-resources}
+\end{nonnormative}
+
+The $\mathit{messageContext}$ following the keyword \lstinline!msgctxt! shall be the full name of the Modelica class (e.g., \lstinline!"Modelica.Blocks.Math.Sin"!) where the text appears.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+The text string which shall be translated is used as $\mathit{messageIdentifier}$ (following the \lstinline!msgid! keyword), and shall contain the original string from the Modelica code.
+% Describe down-quoting and concatenation of Modelica string literals.
+Note that if a \lstinline!msgid! string is given more than once in the same context, all occurrences are translated with the same (last) translation!
+
+\begin{nonnormative}
+The $\mathit{messageTranslation}$ (following the keyword \lstinline!msgstr!) is the translation of $\mathit{messageIdentifier}$ and is typically edited using special tools for translators.
+In the template file this string is empty by definition.
+If this is empty in a language specific file the $\mathit{messageIdentifier}$ may be used instead.
+\end{nonnormative}
+
+\begin{nonnormative}
+Since in Modelica control sequences also start with a backslash and another backslash is used to use sequences literally or to hide double quotes, no change is required here.
+But Modelica allows strings to go over more than one line, gettext does not.
+Therefore, line breaks in Modelica must be encoded with \lstinline!"\n"! for gettext.
+
+In order to avoid long lines in the translated strings (following \lstinline!msgid! or \lstinline!msgstr!), strings may be split into smaller parts given on consecutive lines.
+E.g., the Modelica description
+\begin{lstlisting}
+"A
+B\"C" + "D\nE"
+\end{lstlisting}
+evaluates to:\par
+\hspace{1em}\begin{minipage}{0.5\textwidth}
+A\\
+B"CD\\
+E
+\end{minipage}\par
+which in the translation entry looks like:
+\begin{lstlisting}
+msgid ""
+"A\n"
+"B\"CD\n"
+"E"
+\end{lstlisting}
+\end{nonnormative}
+
+\begin{example}
+Consider a simple sine-source:
+\begin{lstlisting}[language=modelica]
+within MyPackage.Sources;
+model Sine "Sine"
+  parameter Frequency f=2 "Frequency";
+  RealOutput y=sin(2*pi*f); // Relying on imported types
+  /* Could add details. Note that this is not translated */
+  annotation(Icon(graphics={Text(extent={{0,0},{40,40}},
+                        textString="Frequency: %f")}));
+end Sine;
+\end{lstlisting}
+The entries for translating this model into Swedish could look like this:
+\begin{lstlisting}
+#: MyPackage/Sources/package.mo:10126
+#, no-c-format
+msgctxt "MyPackage.Sources.Sine"
+msgid "Sine"
+msgstr "Sinus"
+#: MyPackage/Sources/package.mo:10127
+#, no-c-format
+msgctxt "MyPackage.Sources.Sine"
+msgid "Frequency"
+msgstr "Frekvens"
+#: MyPackage/Sources/package.mo:10129
+#, no-c-format
+msgctxt "MyPackage.Sources.Sine"
+msgid "Frequency: %f"
+msgstr "Frekvens: %f"
+\end{lstlisting}
+\end{example}
+
```

Comment:
Can't we categorized all strings under one context? So we don't have to repeat the context again.

Reply:
That should follow by the gettext specification.  The way I read it, the `msgctxt` belongs to a single `msgid`.

Reply:
Yes. It wasn't 100% clear at first to me, but it says:
> Note that an empty context string and an absent msgctxt line do not mean the same thing.
An absent context only makes sense if it is for each message.